sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -428,9 +428,7 @@ private void mergeFailedReplica(Set<URI> other) {
             return;
         }
 
-        for (URI uri : other) {
-            this.failedReplicas.add(uri);
-        }
+        this.failedReplicas.addAll(other);
```

Comment:
`addAll` can be more performant than a loop on `add`

Reply:
agreed, thanks!